src/Pagination.ts

```diff
@@ -0,0 +1,102 @@
+import { v, w } from '@dojo/widget-core/d';
+import { DNode } from '@dojo/widget-core/interfaces';
+import { theme, ThemeableMixin, ThemeableProperties } from '@dojo/widget-core/mixins/Themeable';
+import WidgetBase from '@dojo/widget-core/WidgetBase';
+import PageLink, { PageLinkProperties } from './pagination/PageLink';
+
+import * as css from './styles/pagination.m.css';
+
+export const PaginationBase = ThemeableMixin(WidgetBase);
+
+export interface PaginationProperties extends ThemeableProperties {
+	page: number;
+	pages: number;
+	status: string;
+	onPageRequest: (pageNumber: number) => void;
+}
+
+@theme(css)
+class Pagination extends PaginationBase<PaginationProperties> {
+	render (): DNode {
+		const {
+			onPageRequest,
+			page,
+			pages,
+			status
+		} = this.properties;
+		const isFirstPage = page === 1;
+		const isLastPage = page === pages;
+		const children: DNode[] = [
+			v('div', {
+				classes: this.classes(css.status)
+			}, [
+				status
+			])
+		];
+
+		if (pages > 1) {
+			const pageLinks: DNode[] = [];
+
+			pageLinks.push(w<PageLinkProperties>(PageLink, { key: '1', disabled: isFirstPage, page: 1, onPageRequest }));
```

Comment:
The generic indicator can be removed on these now/switched to the class. And once the grid goes in, this should probably be moved to a registered widget (and added to the grid registry) rather than using the class directly.